play-validations/memory-footprint/src/main/java/com/google/wear/watchface/dfx/memory/ActiveMemoryFootprintCalculator.java

```diff
@@ -0,0 +1,40 @@
+package com.google.wear.watchface.dfx.memory;
+
+import static com.google.wear.watchface.dfx.memory.WatchFaceDocuments.findSceneNode;
+
+import org.w3c.dom.Document;
+
+import java.util.Map;
+
+/** Computes the memory footprint of a watch face in active. */
+public class ActiveMemoryFootprintCalculator {
```

Comment:
Initially I had a single calculator class, used for both ambient and active, with a few `if` statements to introduce small tweaks in the algorithm based on the variant. However, after the logic changed quite a lot, we have introduced the `AmbientMemoryFootprintCalculator`. By creating this class, I follow the same pattern for active, and delegate a lot of the logic to the `DynamicNodePerConfigurationFootprintCalculator` class, which I intend on using in the ambient calculator as well, in a follow up PR. That class now encapsulates the algorithm used to not have to go over all the possible combinations of user styles, speeding up the whole calculation quite a lot, which needs to be added to the ambient calculator as well.